worker/solver_worker.ml

```diff
@@ -6,6 +6,58 @@ module Context = Context
 module Log = Log
 module Process = Process
 
+module Solver_request = struct
+  open Capnp_rpc_lwt
+  module Worker = Solver_service_api.Worker
+  module Log = Solver_service_api.Solver.Log
+  module Selection = Worker.Selection
+  module Service = Solver_service.Service.Make (Solver_service.Opam_repository)
+
+  type t =
+    ( Service.Epoch.t,
+      Solver_service.Remote_commit.t list )
+    Solver_service.Epoch_lock.t
+
+  let create ~n_workers () =
+    let create_worker commits =
+      let cmd =
+        ( "",
+          [|
+            "solver-service";
+            "--worker";
+            Solver_service.Remote_commit.list_to_string commits;
+          |] )
+      in
+      Lwt_process.open_process cmd
+    in
+    let create commits =
+      Service.Epoch.create ~n_workers ~create_worker commits
+    in
+    Solver_service.Epoch_lock.v ~create ~dispose:Service.Epoch.dispose ()
+
+  let solve t ~switch ~log ~request =
```

Comment:
Not sure to understand, but is this function replacing `spawn_local`? If yes, is the  `switch` required for the local part?

Reply:
Yes the switch is required, to know when a job is canceled, that helps to kill the internal-workers which are concerned.

Reply:
Yes, this replaces `spawn_local` to help accessing the job request switch directly.

Reply:
I see your concerned, It is intentional `(?(switch = Lwt_witch.create ())`, to push giving a switch each time we would like to write a service (Improving the solver-service would be to add a callback in order to use a switch to release resources).

There's 2 types of call in `woker/solver_worker.mli`:
- the function `solve` which create a service using the actual protocol `api/schema.capnp` (_I just forgot removing this function_).
- the function solve_request which not create a service, it uses the pool of solver directly. This where the importance of the switch.

Reply:
How will it work if we want to run a local solver? To be totally clear with that.

Reply:
Yes, you create a `Solver_request.t` and use `Solver_request.solve t ~switch ~log ~request`.

I wonder, if it's better to put this module in its own file.